src/web/mjs/connectors/OnePunchMan.mjs

```diff
@@ -23,22 +22,20 @@ export default class OnePunchMan extends Connector {
     async _getChapters(manga) {
         let request = new Request(manga.id, this.requestOptions);
         let data = await this.fetchDOM(request, '#Chapters_List a');
+
         return data.map(element => {
             return {
-                id: this.getRootRelativeOrAbsoluteLink(element, this.url),
+                id: this.getAbsolutePath(element, this.url),
                 title: element.text.split(', ')[1],
                 language: 'en'
             };
         });
     }
 
     async _getPages(chapter) {
-        let request = new Request(this.url + chapter.id, this.requestOptions);
```

Comment:
May use `new URL(chapter.id, this.url)` instead to handle both cases:
1. The chapter id is a relative path
2. The chapter id is an absolute URL

Reply:
chapter id will always be absolute